metadata/metadata.go

```diff
@@ -832,6 +866,45 @@ func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup) error {
+	val, err := lookup.Lookup(resource.ID())
```

Comment:
This won't work since the model you lookup will be the same one you currently have

Reply:
I could be grossly misunderstanding something here, but if `resource` is the object in memory on which we're calling `Update`, and we already know that a record by the same key exists, isn't the `val` going to be different from `resource`?